common/src/KokkosFFT_utils.hpp

```diff
@@ -209,13 +209,35 @@ auto extract_extents(const ViewType& view) {
   return extents;
 }
 
-template <typename Layout, std::size_t N>
-Layout create_layout(const std::array<int, N>& extents) {
+template <typename IntType, typename InViewType, typename OutViewType>
+std::array<IntType, InViewType::rank> shrank_extents(const InViewType& in,
+                                                     const OutViewType& out) {
+  static_assert(
+      KokkosFFT::Impl::have_same_rank_v<InViewType, OutViewType>,
+      "shrank_extents: InViewType and OutViewType must have the same rank.");
+  static_assert(std::is_integral_v<IntType>,
+                "shrank_extents: IntType must be an integral type");
+  constexpr std::size_t rank = InViewType::rank();
+  using extents_type         = std::array<IntType, rank>;
+
+  extents_type extents;
+  for (std::size_t i = 0; i < rank; i++) {
+    extents.at(i) = std::min(in.extent(i), out.extent(i));
+  }
+  return extents;
+}
+
+template <typename Layout, typename ContainerType>
```

Comment:
Is this change necessary for the current PR ? Does `create_layout` need to be such generic ?

Reply:
Just in case, we want to use `std::vector`? 
Or do you think it is better not to allow that

Reply:
I would generalize when needed

Reply:
Sure, I will keep `std::array<IntType, N>` for the moment then.